packages/server/worker/src/lib/consume/worker-job-rate-limiter.ts

```diff
@@ -122,10 +125,40 @@ export const workerJobRateLimiter = (_log: FastifyBaseLogger) => ({
 
 })
 
-async function getMaxConcurrentJobsPerProject(projectId: ProjectId | undefined): Promise<number> {
-    const storedValue = !isNil(projectId) ? await workerDistributedStore.get(getProjectMaxConcurrentJobsKey(projectId)) : null
-    if (!isNil(storedValue)) {
-        return Number(storedValue)
+const PLAN_CONCURRENT_JOBS_LIMITS: Record<string, number> = {
+    [PlanName.FREE]: 5,
+    [PlanName.APPSUMO_ACTIVEPIECES_TIER1]: 5,
+    [PlanName.APPSUMO_ACTIVEPIECES_TIER2]: 5,
+    [PlanName.APPSUMO_ACTIVEPIECES_TIER3]: 10,
+    [PlanName.APPSUMO_ACTIVEPIECES_TIER4]: 15,
+    [PlanName.APPSUMO_ACTIVEPIECES_TIER5]: 20,
+    [PlanName.APPSUMO_ACTIVEPIECES_TIER6]: 25,
+}
+
+async function getMaxConcurrentJobsPerProject({ projectId, platformId }: GetMaximumConcurrentJovsPerProjectParams): Promise<number> {
+    const keys = [getPlatformPlanNameKey(platformId)]
+    if (!isNil(projectId)) {
+        keys.push(getProjectMaxConcurrentJobsKey(projectId))
     }
-    return workerMachine.getSettings().MAX_CONCURRENT_JOBS_PER_PROJECT
+    const storedValues = await workerDistributedStore.getAll<string>(keys)
+
+    if (!isNil(projectId) && !isNil(storedValues[getProjectMaxConcurrentJobsKey(projectId)])) {
+        return Number(storedValues[getProjectMaxConcurrentJobsKey(projectId)])
+    }
+
+    const platformPlanName = storedValues[getPlatformPlanNameKey(platformId)]
+    if (isNil(platformPlanName)) {
+        return workerMachine.getSettings().MAX_CONCURRENT_JOBS_PER_PROJECT
+    }
+
+    if (workerMachine.getSettings().EDITION === ApEdition.CLOUD && platformPlanName === PlanName.ENTERPRISE) {
+        return 30
```

Comment:
I think this statement should be removed, Just have it part of PLAN_CONCURRENT_JOBS_LIMITS